doc/source/examples/incompressible-flow/3d-taylor-green-vortex/3d-taylor-green-vortex.rst

```diff
@@ -0,0 +1,374 @@
+
+Boundary Conditions
+~~~~~~~~~~~~~~~~~~~
+
+The ``boundary conditions`` subsection establishes the constraints on different parts of the domain:
+
+.. code-block:: text
+
+  subsection boundary conditions
+    set number = 3
+    subsection bc 0
+      set type               = periodic
+      set id                 = 0
+      set periodic_id        = 1
+      set periodic_direction = 0
+    end
+    subsection bc 1
+      set type               = periodic
+      set id                 = 2
+      set periodic_id        = 3
+      set periodic_direction = 1
+    end
+    subsection bc 2
+      set type               = periodic
+      set id                 = 4
+      set periodic_id        = 5
+      set periodic_direction = 2
+    end
+  end
+
+First, the ``number`` of boundary conditions to be applied must be specified. For each boundary condition, the ``id`` of the boundary as well as its ``type`` must be specified. All boundaries are periodic. The ``x-`` (id=0) is periodic with the ``x+``boundary (id=1), the ``y-`` (id=2) is periodic with the ``y+`` boundary (id=3) and so on and so forth. For each periodic boundary condition, the periodic direction must be specified. A periodic direction of ``0`` implies that the normal direction of the wall is the :math:`\mathbf{e}_x` vector, ``1`` implies that it's the :math:`\mathbf{e}_y`.
+
+Physical Properties
+~~~~~~~~~~~~~~~~~~~
+
+The Reynolds number of 1600 is set solely using the kinematic viscosity since the reference velocity is one.
+
+.. code-block:: text
+
+  subsection physical properties
+    set number of fluids = 1
+    subsection fluid 0
+      set kinematic viscosity = 0.000625
+    end
+  end
+
+
+FEM Interpolation
+~~~~~~~~~~~~~~~~~
+
+The results obtained for the Taylor-Green vortex are highly dependent on the numerical dissipation that occurs within the CFD scheme. Generally, high-order methods outperform traditional second-order accurate methods for this type of flow. In the present case, we will investigate the usage of both second and third degree polynomial.
+
+.. code-block:: text
+
+    subsection FEM
+        set velocity order = 2 #3 for Q3
+        set pressure order = 2 #3 for Q3
+    end
+
+Post-processing
+~~~~~~~~~~~~~~~
+
+.. code-block:: text
+
+  subsection post-processing
+    set verbosity                = verbose
+    set calculate enstrophy      = true
+    set calculate kinetic energy = true
+  end
+
+To monitor the kinetic energy and the enstrophy, we set both calculation to ``true`` in the post-processing section.
+
+Simulation Control
+~~~~~~~~~~~~~~~~~~
+
+The ``simulation control`` subsection controls the flow of the simulation. To maximise the temporal accuracy of the simulation, we use a third order ``bdf3`` scheme. Results are written every 2 time-step. To ensure a more adequate visualization of the high-order elements, we set ``subdivision=3``. This will allow Paraview to render the high-order solutions with more fidelity.
```

Comment:
```suggestion
The ``simulation control`` subsection controls the flow of the simulation. To maximise the temporal accuracy of the simulation, we use a third order ``bdf3`` scheme. Results are written every 2 time-steps. To ensure a more adequate visualization of the high-order elements, we set ``subdivision=3``. This will allow Paraview to render the high-order solutions with more fidelity.
```